New package: Ahead v0.1.0

JuliaRegistrator opened the pull request:
- Registering package: Ahead
- Repository: https://github.com/Techtonique/Ahead.jl
- Created by: @thierrymoudiki
- Version: v0.1.0
- Commit: 204df83475f0ecfb601100e42d013273c621d7d8
- Git reference: HEAD
- Description: Univariate and multivariate time series with uncertainty quantification

thierrymoudiki commented:
`Ahead`similar to `Fread`? :open_mouth:

thierrymoudiki commented:
@JuliaRegistrator `Ahead`similar to `Fread`? :open_mouth:

> Your `new package` pull request does not meet the guidelines for auto-merging. Please make sure that you have read the [General registry README](https://github.com/JuliaRegistries/General/blob/master/README.md) and the [AutoMerge guidelines](https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/). The following guidelines were not met:
> 
> * Package name similar to 1 existing package.
>   
>   1. Similar to Fread. Damerau-Levenshtein distance 2 is at or below cutoff of 2.
> 
> Note that the guidelines are only required for the pull request to be merged automatically. However, it is **strongly recommended** to follow them, since otherwise the pull request needs to be manually reviewed and merged by a human.
> 
> After you have fixed the AutoMerge issues, simple retrigger Registrator, which will automatically update this pull request. You do not need to change the version number in your `Project.toml` file (unless of course the AutoMerge issue is that you skipped a version number, in which case you should change the version number).
> 
> If you do not want to fix the AutoMerge issues, please post a comment explaining why you would like this pull request to be manually merged. Then, send a message to the `#pkg-registration` channel in the [Julia Slack](https://julialang.org/slack/) to ask for help. Include a link to this pull request.
> 
> Since you are registering a new package, please make sure that you have also read the package naming guidelines: https://julialang.github.io/Pkg.jl/dev/creating-packages/#Package-naming-guidelines-1
> 
> If you want to prevent this pull request from being auto-merged, simply leave a comment. If you want to post a comment without blocking auto-merging, you must include the text `[noblock]` in your comment. You can edit blocking comments, adding `[noblock]` to them in order to unblock auto-merging.

nilshg commented:
Yes, it is - the difference is only 2 charatcters, which is the cutoff. 

Name similarity isn't an issue though, this is just an automated check to prevent typosquatting. More broadly though did you really intent to register this package with the General registry? It has no documentation so isn't really useful for anyone at this point. I see that this is a wrapper package around an R package that you maintain, maybe have a look at other R wrapper packages like https://github.com/jmboehm/Alpaca.jl to get an idea of how people usually do this in Julia.

thierrymoudiki commented:
Technically, yes. Semantically...

The 'General registry'. Warm welcome. Not sure. The -- existing -- documentation was aimed to be gathered in the same place . On PyPI I don't document. 

> Yes, it is - the difference is only 2 charatcters, which is the cutoff.
> 
> Name similarity isn't an issue though, this is just an automated check to prevent typosquatting. More broadly though did you really intent to register this package with the General registry? It has no documentation so isn't really useful for anyone at this point. I see that this is a wrapper package around an R package that you maintain, maybe have a look at other R wrapper packages like https://github.com/jmboehm/Alpaca.jl to get an idea of how people usually do this in Julia.

nilshg commented:
As I said, this is an automated check to prevent typosquatting, so all it is designed to do is detect technical naming conflicts. As I also said, it shouldn't be an issue in this case - it simply prevents automerging, but as the name is actually not very similar and you have a good reason to stick with this particular name given it wraps an existing package, it won't prevent a manual merge.

How you document the functionality of your package is of course up to you, but you probably should at least allow Julia users to somehow find that documentation, e.g. by stating in the Readme and docstrings to your package where actual documentation (which shows how to use the Julia wrapper) can be found. I see you have a Python wrapper as well which has a good Readme that shows a Python user how to use the package from Python, so that's probably what's needed for the Julia version as well.

thierrymoudiki commented:
I can update the README, and provide a link to the documentation (which mirrors the R version). 

> As I said, this is an automated check to prevent typosquatting, so all it is designed to do is detect technical naming conflicts. As I also said, it shouldn't be an issue in this case - it simply prevents automerging, but as the name is actually not very similar and you have a good reason to stick with this particular name given it wraps an existing package, it won't prevent a manual merge.
> 
> How you document the functionality of your package is of course up to you, but you probably should at least allow Julia users to somehow find that documentation, e.g. by stating in the Readme and docstrings to your package where actual documentation (which shows how to use the Julia wrapper) can be found. I see you have a Python wrapper as well which has a good Readme that shows a Python user how to use the package from Python, so that's probably what's needed for the Julia version as well.

JuliaHubRegistrator committed:
New package: Ahead v0.1.0

UUID: e561dd4c-0aa7-4ba4-aacb-df50ed0db7d4
Repo: https://github.com/Techtonique/Ahead.jl.git
Tree: 32db20e04be37831451c3ab0d1111331edafe980

Registrator tree SHA: 7242ef64be5953fac9bf8b3efa3a7c4d2d44ca09

thierrymoudiki commented:
@nilshg @JuliaRegistrator @JuliaHubRegistrator  Hi, any update on this? Also (this might not be the best place for it), any tips of any existing CI templates for  testing on more platforms? Having various errors depending on the platform.

nilshg commented:
Readme looks fine, but this can't be automerged due to the name similarity. It needs a manual merge, easiest is probably to ping the #pkg-registration channel on the Julia lang slack

thierrymoudiki commented:
What if I don't use slack (which is the case) ? :)

nilshg commented:
Slack isn't required, maintainers also check on PRs without them being bumped on Slack. In this case however I have put a message in the Slack channel to hopefully speed things up.

thierrymoudiki commented:
I appreciate it. Cheers

giordano commented:
[The package can't be loaded](https://github.com/JuliaRegistries/General/actions/runs/7652346290/job/20851927611?pr=98771#step:16:14391):
```
ERROR: LoadError: REvalError: Error in library(ahead, lib.loc = ".") : 
  there is no package called 'ahead'
```
Please read https://github.com/JuliaRegistries/General/blob/8edde9cccb3bc49ced997525a7b607ef61aac5f7/README.md#my-package-fails-to-load-because-it-needs-proprietary-softwareadditional-setup-to-work-what-can-i-do, but the best solution would really be to make this actually work.

thierrymoudiki commented:
@giordano this is a workaround for cases when the package can't be installed by a 'non-staff' member. GitHub Actions tests won't pass without it. Or maybe you'd suggest something else?

goerz commented:
> Slack isn't required, maintainers also check on PRs without them being bumped on Slack.

Do they really? Not my experience ;-) But you might be able to get a non-maintainer post on your behalf on Slack. As @nilshg did, in this case.

giordano commented:
> Or maybe you'd suggest something else?

Yes, reading https://github.com/JuliaRegistries/General/blob/8edde9cccb3bc49ced997525a7b607ef61aac5f7/README.md#my-package-fails-to-load-because-it-needs-proprietary-softwareadditional-setup-to-work-what-can-i-do.  Or making loading the R package work.

thierrymoudiki commented:
> > Or maybe you'd suggest something else?
> 
> Yes, reading https://github.com/JuliaRegistries/General/blob/8edde9cccb3bc49ced997525a7b607ef61aac5f7/README.md#my-package-fails-to-load-because-it-needs-proprietary-softwareadditional-setup-to-work-what-can-i-do. Or making loading the R package work.

All I see below is green checks.

goerz commented:
“2 failing and 9 successful checks”

???

giordano commented:
> All I see below is green checks.

I don't 
![image](https://github.com/JuliaRegistries/General/assets/765740/19412dfe-a13e-42ff-93bc-4c29009fe1e1)

Also, quoting from https://github.com/JuliaRegistries/General/pull/98771#issuecomment-1888911404

> I was not able to load the package (i.e. import Ahead failed). See the CI logs for details.

thierrymoudiki commented:
> “2 failing and 9 successful checks”
> 
> ???

@goerz failing to merge, that means. All the others tests do pass on Ubuntu. Read carefully.

thierrymoudiki commented:
> > All I see below is green checks.
> 
> I don't ![image](https://private-user-images.githubusercontent.com/765740/300791027-19412dfe-a13e-42ff-93bc-4c29009fe1e1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MTQ4NTcsIm5iZiI6MTcwNjYxNDU1NywicGF0aCI6Ii83NjU3NDAvMzAwNzkxMDI3LTE5NDEyZGZlLWExM2UtNDJmZi05M2JjLTRjMjkwMDlmZTFlMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEzMFQxMTM1NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NTIwN2I0YmQ2YmFmYmQ0YjY4YjRmZGEzY2FmYzJjNTE2NzNlYWI0ZjU5ZTg5MTY5ZDUxN2JkZjJjZTE3ODRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.72EXzGkwu_9qvprqXl0kUJOlXhp4sTkXzupqYfWL3uc)
> 
> Also, quoting from [#98771 (comment)](https://github.com/JuliaRegistries/General/pull/98771#issuecomment-1888911404)
> 
> > I was not able to load the package (i.e. import Ahead failed). See the CI logs for details.

@giordano indeed, on some platforms it fails randomly. But not on your official template. Quoting 'can't be automerged due to the name similarity'.

goerz commented:
> Read carefully

Good luck with your registration. I’m out.

thierrymoudiki commented:
> > Read carefully
> 
> Good luck with your registration. I’m out.

@goerz Ha ha. Honesty is disturbing right? :smiley_cat: Bye

Seelengrab commented:
This is what I see:

![image](https://github.com/JuliaRegistries/General/assets/11753998/cae25049-389c-41d6-8b63-75f3374510b1)

The log is also the AutoMerge CI job (see [here](https://github.com/JuliaRegistries/General/actions/runs/7652346290/job/20851927611?pr=98771#step:16:14391)), which tries to ultimately install the package too. Note that the original message by the bot also reported that it can't load the package:

> * I was not able to load the package (i.e. import Ahead failed). See the CI logs for details.

giordano commented:
> Honesty is disturbing right? 😺 Bye

@thierrymoudiki This isn't working very well, please refrain from further comments like that.

> indeed, on some platforms it fails randomly.

It's not random, the R package isn't installed.  I'm not going to repeat it a fourth time as I'm not interested in wasting more time in this discussion, please read https://github.com/JuliaRegistries/General/blob/8edde9cccb3bc49ced997525a7b607ef61aac5f7/README.md#my-package-fails-to-load-because-it-needs-proprietary-softwareadditional-setup-to-work-what-can-i-do.

thierrymoudiki commented:
> > Honesty is disturbing right? 😺 Bye
> 
> @thierrymoudiki This isn't working very well, please refrain from further comments like that.
> 
> > indeed, on some platforms it fails randomly.
> 
> It's not random, the R package isn't installed. I'm not going to repeat it a fourth time as I'm not interested in wasting more time in this discussion, please read https://github.com/JuliaRegistries/General/blob/8edde9cccb3bc49ced997525a7b607ef61aac5f7/README.md#my-package-fails-to-load-because-it-needs-proprietary-softwareadditional-setup-to-work-what-can-i-do.

@giordano More factual details here ;) : https://github.com/Techtonique/Ahead.jl/actions/runs/7710059555

nilshg commented:
Not sure why this thread has disintegrated to this extent, but just to clarify:

>  failing to merge AUTOMATICALLY, that means. All the others tests do pass on Ubuntu. Read carefully.     

"All the other tests" are just the CI of the General registry (which this is a PR to), *not* the CI of your package. In short, all they tell us is that the General registry is not broken if `Ahead` is added to it. 

The one check that is failing is automerge which as you rightly point out is checking whether the package can automatically be merged or needs human intervention. Here as discussed earlier the check failed due to name similarity, which is not an issue for this package so that it could be merged manually.

However if you look into the failed action as @Seelengrab shows above what you'll see is that the check failed because the package can't be loaded, i.e. it is not currently working. This is also confirmed by your own CI, which is the relevant one here:

![image](https://github.com/JuliaRegistries/General/assets/7264491/d366962a-aa25-429e-823b-1199e11668df)

This doesn't look like random failures, but rather consistent failures to do with an apparent issue with how the package is built. This should be fixed before a merge to General - all that the automerge action is doing here is checking that a package registered in General can actually be sucessfully installed, which is imho a low bar.

thierrymoudiki commented:
More facts on the facts, just in case. Windows, macOS, Ubuntu, Julia 1.6 and 1.10: 

https://github.com/Techtonique/Ahead.jl/actions/runs/7711922129/job/21018276042

The most accurate answer for all of us wasn’t [‘it doesn’t work on my machine’](https://builtin.com/software-engineering-perspectives/it-works-on-my-machine-bug)
or ‘you can’t read’ , but rather ‘it doesn’t work **on Windows right now**’. Or maybe... **testing more OSes in the official CI template**.

nilshg commented:
Looks like you've sorted the installation issues - if you trigger the registration bot again (without changing version number or anything else) the PR should update and the automerge check should be re-run.

goerz commented:
In this case, they might actually be better off updating the version and submitting a new registration. Even if the automerge issue is solved, there are a lot of blocking comments on this one. Mosè isn't going to be back to unblock his, and I'm not sure another maintainer is going to merge this manually (future tip: don't be rude to registry maintainers when you need their help).

I can override the name similarity check on a new submission.
